docs/ref/contrib/postgres/fields.txt

```diff
@@ -358,6 +358,19 @@ field. It uses the SQL operator ``@>``. For example::
     >>> Dog.objects.filter(data__contains={'breed': 'collie'})
     <QuerySet [<Dog: Meg>]>
 
+If you want to search for objects where the title contains a substring instead of an
+exact match, you'll have to cast your JSON field to text with :class:`~django.db.models.functions.Cast`
+before hand. For example::
+
+    >>> Dog.objects.create(name='Rufus', data={'hobby': 'Hanging around with cats'})
+
+    >>> from django.db import models
+    >>> from django.db.models.functions import Cast
+    >>> Dog.objects.annotate(
+    ...     text=Cast('data', models.TextField())
```

Comment:
Based on further investigation it looks like what we really need here is support for the `->>` operator as `(lhs -> key)::text` is the JSON representation of the string (`('"foo"'::jsonb)::text` -> `'"foo"'`).

I believe this should be done by introducing a new expression for this purpose and documenting it. In the meantime we could document using `Func('data', template="%(expressions)s ->> 'hobby'", output_field=models.TextField())`.